src/libraries/System.Formats.Nrbf/src/System/Formats/Nrbf/SerializationRecordId.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Formats.Nrbf.Utils;
+using System.IO;
+using System.IO.Hashing;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace System.Formats.Nrbf;
+
+/// <summary>
+/// The ID of <see cref="SerializationRecord" />.
+/// </summary>
+public readonly struct SerializationRecordId : IEquatable<SerializationRecordId>
```

Comment:
See my comment about bitwise-equality: https://github.com/dotnet/runtime/pull/103232/files/a171648cfff83059d12b0021fc162bea9d67de5b#r1635270940

Reply:
@adamsitnik, @GrabYourPitchforks overriding `GetHashCode` or `Equals` makes this id not bitwise equatable from the runtime's perspective. This makes searching for existing ids in arrays suboptimal. Is there a reason we can't expose the underlying value?

Reply:
> This makes searching for existing ids in arrays suboptimal.

But with my latest changes (https://github.com/dotnet/runtime/commit/daa743115d02e9142101f41985ec4f808fbdb6a6) we don't need to do that anymore?

Reply:
cc @bartonjs

Reply:
@adamsitnik we don't for that specific usage, but other usages would hurt.

@jkotas I'm not sure what you mean? Can you point out the general bitwise equatable?

Reply:
The issue that I have linked in my comment has the design discussion about it.

Reply:
@jkotas, sorry I misunderstood your comment. I thought there was some sort of internal back door, not that we needed to add links to the existing issue. :)